src/MongoDB.Driver.Core/Core/Operations/BulkUnmixedWriteOperationBase.cs

```diff
@@ -198,14 +198,30 @@ private void EnsureCollationIsSupportedIfAnyRequestHasCollation(RetryableWriteCo
         private BulkWriteBatchResult ExecuteBatch(RetryableWriteContext context, Batch batch, CancellationToken cancellationToken)
         {
             var operation = CreateBatchOperation(batch);
-            var operationResult = RetryableWriteOperationExecutor.Execute(operation, context, cancellationToken);
+            BsonDocument operationResult;
+            try
+            {
+                operationResult = RetryableWriteOperationExecutor.Execute(operation, context, cancellationToken);
+            }
+            catch (MongoWriteConcernException exception) when (exception.IsWriteConcernErrorOnly())
+            {
+                operationResult = exception.Result;
```

Comment:
For purposes of batch processing turn any MongoWriteException that was thrown ONLY because of a writeConcernError back into a result which is what the batch processing logic is expecting.

Reply:
If it's only a write concern error we continue doing exactly what we used to do.

The batch processing logic decides how to handle the write concern error, just like it used to.